src/transformState.js

```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  let ACTION_REQUIRED = '';
+  
+
+  for (const action of Object.values(actions)) {
+    ACTION_REQUIRED = action['type'];
+
+    switch (ACTION_REQUIRED) {
+      case 'addProperties':
+        Object.assign(state, action['extraData']);
+        break;
+
+      case 'removeProperties':
+        for (const parameters of action['keysToRemove']) {
```

Comment:
Instead of accessing object properties with action['keysToRemove'], use dot notation (action.keysToRemove). Ensure that the 'keysToRemove' property is correct as per the task description.